src/Conversion/ONNXToTorch/NN/ElementwiseOp.cpp

```diff
@@ -0,0 +1,151 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===------- SqrtOp.cpp - ONNX Op Transform ------------------===//
+//
+// Copyright 2022, Helprack LLC.
+//
+// =============================================================================
+//
+// This file lowers most unary operators from torch to onnx using a template
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/Conversion/ONNXToTorch/ONNXToTorchCommon.hpp"
+
+using namespace mlir;
+using namespace mlir::torch;
+using namespace mlir::torch::Torch;
+
+// Element-wise unary ops lowering to Krnl dialect.
+//===----------------------------------------------------------------------===//
+template <typename ONNXUnaryOp, typename TorchUnaryOp>
+struct ONNXToTorchElementwiseUnaryOpLowering : public ConversionPattern {
+  ONNXToTorchElementwiseUnaryOpLowering(TypeConverter &typeConverter, MLIRContext *ctx)
+      : ConversionPattern(
+            typeConverter, ONNXUnaryOp::getOperationName(), 1, ctx) {}
+  LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const final {
+    ONNXUnaryOp unaryOp = llvm::dyn_cast_or_null<ONNXUnaryOp>(op);
+
+    assert(unaryOp && "Expecting op to have a strong type");
+
+    Location loc = unaryOp.getLoc();
+    Value operand = unaryOp.input();
+    mlir::MLIRContext *context =  unaryOp.getContext();
+
+    auto operandTensorType = operand.getType().template dyn_cast<TensorType>();
+    TensorType resultTensorType = unaryOp.getResult().getType().template dyn_cast<TensorType>();
+
+    auto operandType = Torch::ValueTensorType::get(context,
+                                           operandTensorType.getShape(),
+                                           operandTensorType.getElementType());
```

Comment:
It seems like we do this quite often, here and in conversions for other ops. Would it make sense to factorize this into a free standing function somewhere?
```c++
Torch::ValueTensorType toTorchType(mlir::MLIRContext ctx, Type t) {
   auto type = t.template dyn_cast<TensorType>();
   return Torch::ValueTensorType(ctx, type.getShape(), type.getElementType());
}
// then use like
auto operandType = toTorchType(context, unaryOp.getOperand().getType());
auto resultType = toTorchType(context, unaryOp.getResult().getType());

Reply:
Done

Reply:
Thanks